resource/configmap/spec.go

```diff
@@ -0,0 +1,20 @@
+package configmap
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type StateGetter interface {
+	// GetCurrentState returns a current state of the system for the given
+	// carnation of the observed Kubernetes object.
```

Comment:
```suggestion
	// incarnation of the observed Kubernetes object.
```

Pretty sure is incarnation. Nice use of the word!

Maybe its stating the obvious but sometimes we want to create a single configmap other times its multiple. Should that be mentioned in the godoc? Could be useful for future people integrating the resource.

Reply:
IDK. This is fine to produce single CM as a current/desired state. I can add a sentence here.

Reply:
Added this
```
	// incarnation of the observed Kubernetes object. The state consists of
	// multiple objects but it is fine to return a slice of single object
	// if only one object is managed by this resource.
```